code/modules/mob/mob_helpers.dm

```diff
@@ -462,6 +462,8 @@
 			else
 				user.visible_message("[user] has fixed some of the [dam ? "dents on" : "burnt wires in"] [H]'s [parse_zone(affecting.body_zone)].", \
 					"<span class='notice'>You fix some of the [dam ? "dents on" : "burnt wires in"] [H == user ? "your" : "[H]'s"] [parse_zone(affecting.body_zone)].</span>")
+			if(((affecting.brute_dam <= 0 && brute_heal) && ((!H.is_bleeding()) && H.has_mechanical_bleeding())) || (affecting.burn_dam <= 0 && burn_heal)) //might be bad to have this many expressions in a single if....
```

Comment:
I agree with the comments,  Iwould split the burn and brute part of this expression